src/clusterfuzz/_internal/bot/tasks/utasks/__init__.py

```diff
@@ -144,10 +182,51 @@ def uworker_bot_main():
   return 0
 
 
+def batch_record_finished_metrics(start, utask_module, job_type, subtask):
```

Comment:
Nit: This sounds like a batch operation where you're recording finished metrics multiple times in a single call. The one below sounds like we're queueing a task to record metrics later. I suggest defining mode constants (or an enum) for `"batch"` and `"queue"` and passing those as arguments to `record_finished_metrics` directly.